core/app/models/spree/payment_method/store_credit.rb

```diff
@@ -0,0 +1,115 @@
+module Spree
+  class PaymentMethod::StoreCredit < PaymentMethod
+
+    def payment_source_class
+      ::Spree::StoreCredit
+    end
+
+    def can_capture?(payment)
+      ['checkout', 'pending'].include?(payment.state)
+    end
+
+    def can_void?(payment)
+      payment.pending?
+    end
+
+    def authorize(amount_in_cents, store_credit, gateway_options = {})
+      if store_credit.nil?
+        ActiveMerchant::Billing::Response.new(false, Spree.t('store_credit_payment_method.unable_to_find'), {}, {})
+      else
+        action = -> (store_credit) {
```

Comment:
Shadowing outer local variable - `store_credit`.<br>Avoid using {...} for multi-line blocks.